content/docs/reference-react-dom-server.md

```diff
@@ -40,47 +40,91 @@ The following methods can be used in the environments that don't support streams
 ReactDOMServer.renderToPipeableStream(element, options)
 ```
 
-Render a React element to its initial HTML. Returns a stream with a `pipe(res)` method to pipe the output and `abort()` to abort the request. Fully supports Suspense and streaming of HTML with "delayed" content blocks "popping in" via inline `<script>` tags later. [Read more](https://github.com/reactwg/react-18/discussions/37)
+Render a React element to its initial HTML without interaction. Returns a [Node.js stream](https://nodejs.dev/learn/nodejs-streams) with a `pipe(res)` method to pipe the output and `abort()` to abort the request.
 
-If you call [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on a node that already has this server-rendered markup, React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.
+This method fully supports Suspense, and offers two approaches for handling suspended content. The first lets you stream content blocks as they become ready. The second waits until **all** suspended content blocks are ready before sending a response.
+
+#### Suspense approach one: Initial shell with streamed content blocks
```

Comment:
Describing it as two approaches sounds like you can pick either one. In practice you should pick the first one for everything except static generation and maybe crawlers (although some crawlers can handle JS too). So I think we'd want to describe this based on the use case rather than either/or. Also, describing them separately makes it hard to explain how to switch the approach based on condition like `isCrawler`.

Reply:
Yeah I agree. I’m trying to find a term of art for what this is. “When you can’t support client-side JavaScript”

Reply:
@gaearon I’ve updated the documentation to follow more how you say, where `onShellReady` is the default and can be opted-into `onAllReady` based on a `isCrawler` condition. Not sure if it’s clear how I’ve done it, please let me know :)